packages/gasket-plugin-lifecycle/index.js

```diff
@@ -26,8 +26,9 @@ async function resolve(root, name) {
     }
   }
 
+  const reg = /.*^((?!(\.test|\.spec).js).)*\.js$/;
```

Comment:
This assumes that the test files are name with the convention of `XXXXX.test.js` or `XXXX.spec.js`. It might be better to make this the default regex for a configurable value.